packages/cerebral/src/Module.js

```diff
@@ -1,35 +1,24 @@
 class Module {
-  constructor (path, module, controller) {
-    this.name = path.slice().pop()
-    this.path = path
-    this.controller = controller
-    this.modules = {}
+  constructor (controller, path, moduleDescription) {
+    const moduleStub = {
+      controller,
+      path: path.join('.'),
+      name: path.slice().pop()
+    }
 
-    const moduleDescription = typeof module === 'function' ? module(this) : module
+    const module = typeof moduleDescription === 'function' ? moduleDescription(moduleStub) : moduleDescription
```

Comment:
Do not understand that this works? You have to use `new moduleDescription(moduleStub)` for classes to work. So check if moduleDescription is a a basic object, if not use `new`... does not matter if it is a class or a plain function as you return a complex object which will be used even if you do a "new" on it...